Core/include/Acts/Clusterization/HkClusterization.ipp

```diff
@@ -0,0 +1,240 @@
+}
+
+template <typename T, typename U>
+constexpr void static_check_cluster_type() {
+  constexpr bool has_fns = cluster_type_has_required_functions<T, U>();
+  static_assert(has_fns,
+                "Cluster type should have the following function: "
+                "'void cluster_add_cell(Cluster&, const Cell&)'");
+}
+
+// Comparator function object for cells, column-wise ordering
+template <typename Cell>
+struct Compare {
+  bool operator()(const Cell& c0, const Cell& c1) const {
+    int row0 = get_cell_row(c0);
+    int row1 = get_cell_row(c1);
+    int col0 = get_cell_column(c0);
+    int col1 = get_cell_column(c1);
+    return (col0 == col1) ? row0 < row1 : col0 < col1;
+  }
+};
+
+// Simple wrapper around boost::disjoint_sets. In theory, could use
+// boost::vector_property_map and use boost::disjoint_sets without
+// wrapping, but it's way slower
+class DisjointSets {
+ public:
+  DisjointSets(size_t initial_size = 128)
+      : m_global_id(1),
+        m_size(initial_size),
+        m_rank(m_size),
+        m_parent(m_size),
+        m_ds(&m_rank[0], &m_parent[0]) {}
+
+  Label make_set() {
+    // Empirically, m_size = 128 seems to be good default. If we
+    // exceed this, take a performance hit and do the right thing.
+    while (m_global_id >= m_size) {
+      m_size *= 2;
+      m_rank.resize(m_size);
+      m_parent.resize(m_size);
+      m_ds = boost::disjoint_sets<size_t*, size_t*>(&m_rank[0], &m_parent[0]);
+    }
+    m_ds.make_set(m_global_id);
+    return static_cast<Label>(m_global_id++);
+  }
+
+  void union_set(size_t x, size_t y) { m_ds.union_set(x, y); }
+  Label find_set(size_t x) { return static_cast<Label>(m_ds.find_set(x)); }
+
+ private:
+  size_t m_global_id;
+  size_t m_size;
+  std::vector<size_t> m_rank;
+  std::vector<size_t> m_parent;
+  boost::disjoint_sets<size_t*, size_t*> m_ds;
+};
+
+// Cell collection logic
+// TODO: add template parameter to extend matching logic?
+template <typename Cell, typename Connect>
+int get_connections(typename std::vector<Cell>::iterator it,
+                    std::vector<Cell>& set, Connect connect,
+                    std::array<Label, 4>& seen) {
+  int nconn = 0;
+  seen[0] = seen[1] = seen[2] = seen[3] = NO_LABEL;
+  typename std::vector<Cell>::iterator it_2{it};
+
+  while (it_2 != set.begin()) {
+    it_2 = std::prev(it_2);
+
+    ConnectResult cr = connect(*it, *it_2);
+    if (cr == NO_CONN_STOP)
+      break;
+    if (cr == NO_CONN)
+      continue;
+    if (cr == CONN) {
+      seen[nconn++] = get_cell_label(*it_2);
+      if (nconn == 4)
+        break;
+    }
+  }
+  return nconn;
+}
+
+}  // namespace internal
+
+template <typename Cell>
+ConnectResult DefaultConnect<Cell>::operator()(const Cell& a, const Cell& b) {
+  int delta_row = std::abs(get_cell_row(a) - get_cell_row(b));
+  int delta_col = std::abs(get_cell_column(a) - get_cell_column(b));
+  // Iteration is column-wise, so if too far in column, can
+  // safely stop
+  if (delta_col > 1)
```

Comment:
Also here, in ACTS, we try to use brackets wherever needed, i.e. this would read:
```c++
if (...){
...
}
```

I know this is just naming, but we try to keep a consistent code base (I didn't bother what style conventions, I would just like to keep one).

Reply:
Done